### PR TITLE
Fix retain cycle with transaction operations

### PR DESCRIPTION
**Pull request by bolsinga**
Add unit tests that help find cycles. `-testWeakWithSingleOperation` fails without the code fix applied.

### CODE REVIEW

**Comment by garrettmoon** [comment]
🥇

File: Tests/ASTransactionTests.mm [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+  @autoreleasepool {
+    CALayer *layer = [[CALayer alloc] init];
+    _ASAsyncTransaction *transaction = layer.asyncdisplaykit_asyncTransaction;
+
+    [transaction addOperationWithBlock:^id<NSObject> _Nullable{
+      return nil;
+    } priority:1
+                                 queue:dispatch_get_main_queue()
+                            completion:^(id  _Nullable value, BOOL canceled) {
+                              ;
```
Comment (mikezucc) [inline Tests/ASTransactionTests.mm]: can this just be a nil block?

**Comment by pinterest-ios** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Please ensure license is correct for ASTransactionTests.mm:</p>
<pre><code>//
//  ASTransactionTests.mm
//  Texture
//
//  Copyright (c) Pinterest, Inc.  All rights reserved.
//  Licensed under Apache 2.0: http://www.apache.org/licenses/LICENSE-2.0
</code></</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by pinterest-ios** [comment]
🚫 CI failed with [log](https://gist.githubusercontent.com/pinterest-ios/cc1aad6027ac2043672f41f74fa9929c/raw/7ae43493d094e005e84f66551ee6249afa1add8f/log.txt)

**Comment by pinterest-ios** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Please ensure license is correct for ASTransactionTests.mm:</p>
<pre><code>//
//  ASTransactionTests.mm
//  Texture
//
//  Copyright (c) Pinterest, Inc.  All rights reserved.
//  Licensed under Apache 2.0: http://www.apache.org/licenses/LICENSE-2.0
</code></</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Pull request merged**